src/posit/connect/external/databricks.py

```diff
@@ -7,24 +7,28 @@
 These APIs are provided as a convenience and are subject to breaking changes:
 https://github.com/databricks/databricks-sdk-py#interface-stability
 """
-
-import abc
+from __future__ import annotations
 
 import requests
-from typing_extensions import Callable, Dict, Optional
+from typing_extensions import TYPE_CHECKING, Dict, Optional
 
-from .._utils import is_local
+from .._utils import is_local, is_connect, is_workbench
 from ..client import Client
 from ..oauth import Credentials
 
+if TYPE_CHECKING:
+    from databricks.sdk.core import Config
+    from databricks.sdk.credentials_provider import (
+            CredentialsStrategy,
+            CredentialsProvider
+    )
+
 POSIT_OAUTH_INTEGRATION_AUTH_TYPE = "posit-oauth-integration"
 POSIT_LOCAL_CLIENT_CREDENTIALS_AUTH_TYPE = "posit-local-client-credentials"
-
-# The Databricks SDK CredentialsProvider == Databricks SQL HeaderFactory
-CredentialsProvider = Callable[[], Dict[str, str]]
+POSIT_WORKBENCH_AUTH_TYPE = "posit-workbench"
 
 
-class CredentialsStrategy(abc.ABC):
+class CredentialsStrategyWrapper(CredentialsStrategy):
```

Comment:
Need to revisit this. I'd rather not depend on the databricks sdk but this violates the `TYPE_CHECKING` import above.